docs/parameters.md

```diff
@@ -0,0 +1,234 @@
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      type: json # Explicitly tells that the value is JSON which should be validated and passed as a JSON object
+      value: >
+          {
+            "attributes": [
+                {
+                  "type": "a",
+                  "A": "1",
+                },
+                {
+                  "type": "b",
+                  "B": "2",
+                },
+            ],
+            "labels": ["foo", "bar"]
+          }
+```
+
+If the JSON object is stored in a `Secret` or a `ConfigMap` key, the combination of `valueFrom` and `type: json` fields 
+allows to declare this as well:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      type: json # Explicitly tells that the value is JSON which should be validated and passed as a JSON object
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: password
+```
+
+### Passing all parameters from a single source at once
+
+In addition to support for explicitly specifying every parameter value fetched from a particular source, Service Catalog
+also supports passing multiple parameters at once using a `parametersFrom` field (similar to how `envFrom` works for `Pod`s).
+Instead of specifying a reference to a single `Secret` 
+(or `ConfigMap`) key, you just set the reference to the `Secret` (or `ConfigMap`), and all keys will be automatically
+processed as separate parameters.
+
+In other words, instead of 
+```yaml
+  ...
+  parameters:
+    - name: key1
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key1
+    - name: key2
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key2
+     ...
+```
+you can just set it once in a `parametersFrom` field:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: mysecret
+```
+
+You can also pass several `Secret`s and/or `ConfigMap`s at once:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: secret1
+    - secretRef:
+        name: secret2
+    - configMapRef:
+        name: configmap1
+```
+
+Parameter values could be passed as JSON objects, the same way as for [individual parameters](#passing-parameter-value-as-a-json-object):
+```yaml
+  ...
+  parametersFrom:
+    - type: json
+      secretRef:
+        name: mysecret
+```
+
+### Conflict resolution
+
+When a key exists in multiple sources defined in `parametersFrom` or `parameters` field, the value associated with 
+the last source will take precedence.
+Values defined in `parameters` with a duplicate key will take precedence over `parametersFrom`.
```

Comment:
It is worth mentioning that this is consistent with the environment API for containers.